tree/readspeed/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.
+# All rights reserved.
+#
+# For the licensing terms see $ROOTSYS/LICENSE.
+# For the list of contributors see $ROOTSYS/README/CREDITS.
+
+############################################################################
+# CMakeLists.txt file for building ROOT readspeed package
+# @author Bertrand Bellenot CERN
+############################################################################
+
+ROOT_OBJECT_LIBRARY(ReadSpeed
+  src/ReadSpeed.cxx
+  src/ReadSpeedCLI.cxx
+)
+
+add_dependencies(ReadSpeed RIO Tree TreePlayer)
+
+target_include_directories(ReadSpeed PRIVATE
+  ${CMAKE_SOURCE_DIR}/io/io/inc
+  ${CMAKE_SOURCE_DIR}/tree/tree/inc
+  ${CMAKE_SOURCE_DIR}/tree/treeplayer/inc
+  ${CMAKE_SOURCE_DIR}/tree/dataframe/inc
+  ${CMAKE_SOURCE_DIR}/core/foundation/inc
+  ${CMAKE_SOURCE_DIR}/core/imt/inc
+  ${CMAKE_SOURCE_DIR}/math/mathcore/inc
+  ${CMAKE_SOURCE_DIR}/math/matrix/inc
+  ${CMAKE_SOURCE_DIR}/math/vecops/inc
+  ${CMAKE_SOURCE_DIR}/hist/hist/inc
+  ${CMAKE_BINARY_DIR}/ginclude
```

Comment:
@bellenot do we really need to add a directory in `CMAKE_BINARY_DIR` to the include directories?

Also `math/matrix`, `math/vecops` and `/tree/dataframe` seem unnecessary. And if we declare which libraries we depend on with `target_link_libraries` doesn't cmake automatically add their include directories to our include directories?

Reply:
Not sure about `${CMAKE_BINARY_DIR}/ginclude` (I think I copied that from somewhere else). For the others, I added them when needed, i.e. when the compilation failed due too missing includes. I should investigate more.
I just gave a quick recipe how to add `readspeed` in ROOT, but if you want I can investigate more an find another solution (there are several ways of doing it). I will invest more time in it...

Reply:
@dkpoult as per the discussion above, can you please try out slimming down the list of include directories to only the ones we actually need?

Reply:
@eguiraud managed to thin this out and remove the line mentioned in the other comment